### DISCUSSION TITLE
About "primalMinResTol"

### DISCUSSION THREAD

**Discussion by hung3vi** [mdolab/dafoam#695] [General]
I have a question about "primalMinResTol" Is that how to avoid increasing the tolerance of the primal when increasing the density of the mesh? I am currently working on an aerodynamic optimization with Naca 2412. When the mesh has the same resolution as the one provided in the tutorial, I can set the primalMinResTol at 1e-8 or even 1e-9. However, when I increased the mesh (about 6 times denser than the tutorial one), I had to set the value up to 1e-5. and the "inf_pr" and "inf_du" do not converge well.

I will attach the runScript, genAirfoilMesh, and the opt_IPOPT below.

Thank you for reading and have a nice day!

[opt_IPOPT.txt](https://github.com/user-attachments/files/17336329/opt_IPOPT.txt)
[runScript_v2.txt](https://github.com/user-attachments/files/17336334/runScript_v2.txt)

[genAirFoilMesh.txt](https://github.com/user-attachments/files/17336336/genAirFoilMesh.txt)

**Comment by friedenhe** [comment]
Please still use primalMinResTol=1e-8. If the flow does not converge well, the optimization won't. And you can set primalMinResTolDiff = 1e4 to prevent DAFoam from thinking a 1e-5 flow convergence is a failed run.
